homeassistant/components/homekit/type_media_players.py

```diff
@@ -371,19 +386,25 @@ def set_input_source(self, value):
     def set_remote_key(self, value):
         """Send remote key value if call came from HomeKit."""
         _LOGGER.debug("%s: Set remote key to %s", self.entity_id, value)
-        service = MEDIA_PLAYER_KEYS.get(value)
-        if service:
+        key_name = MEDIA_PLAYER_KEYS.get(value)
+        if key_name:
             # Handle Play Pause
-            if service == SERVICE_MEDIA_PLAY_PAUSE:
+            if key_name == KEY_PLAY_PAUSE:
                 state = self.hass.states.get(self.entity_id).state
                 if state in (STATE_PLAYING, STATE_PAUSED):
                     service = (
                         SERVICE_MEDIA_PLAY
                         if state == STATE_PAUSED
                         else SERVICE_MEDIA_PAUSE
                     )
-            params = {ATTR_ENTITY_ID: self.entity_id}
-            self.call_service(DOMAIN, service, params)
+                else:
+                    service = SERVICE_MEDIA_PLAY_PAUSE
+                params = {ATTR_ENTITY_ID: self.entity_id}
+                self.call_service(DOMAIN, service, params)
+
+            self.hass.bus.fire(
```

Comment:
If we call a service we should probably not fire an event. There will be a state change event as a result of the service and we shouldn't fire multiple events from the same cause.